File: frontend-admin/src/types/knowledge.ts
export type KnowledgeUploadStatus = 'pending' | 'processing' | 'completed' | 'failed';

export interface KnowledgeUpload {
    id: string;
    filename: string;
    content_type?: string | null;
    file_size?: number | null;
    uploaded_by?: string | null;
    status: KnowledgeUploadStatus;
    error_message?: string | null;
    created_at: string;
    updated_at?: string | null;
    completed_at?: string | null;
    articles_count: number;
}

File: backend/app/schemas/knowledge.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel

from app.models.knowledge import KnowledgeUploadStatus

class KnowledgeUploadResponse(BaseModel):
    id: UUID
    filename: str
    content_type: Optional[str] = None
    file_size: Optional[int] = None
    uploaded_by: Optional[str] = None
    status: KnowledgeUploadStatus
    error_message: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    articles_count: int = 0

    class Config:
        from_attributes = True


class KnowledgeUploadListResponse(BaseModel):
    items: list[KnowledgeUploadResponse]
    totalItems: int
    page: int
    pageSize: int
    totalPages: int

class KnowledgeImportResponse(BaseModel):
    message: str
    upload_id: UUID
    stats: dict
    status: KnowledgeUploadStatus
